.github/workflows/build.yml

```diff
@@ -100,12 +100,15 @@ jobs:
       id: run_tests
       run: |
         export TERM=dumb
-        status=0
-        script -qe -c "python ci/run-tests.py --headless --keep-status \"$DF_FOLDER\"" || status=$((status + 1))
-        python ci/check-rpc.py "$DF_FOLDER/dfhack-rpc.txt" || status=$((status + 2))
-        mkdir -p artifacts
-        cp "$DF_FOLDER"/test*.json "$DF_FOLDER"/*.log artifacts || status=$((status + 4))
-        exit $status
+        if build-ci/library/tests/test-library; then
```

Comment:
Could `build-ci/library/tests/test-library` be a separate step instead? I think it's conceptually separate from this step, and the `if` condition makes things somewhat messy. A separate step would also avoid the need for `exit 1` at the end.